pkg/agent/openflow/pipeline.go

```diff
@@ -2072,209 +2063,163 @@ func (c *client) dnsPacketInFlow(conjunctionID uint32) binding.Flow {
-// to mark the Service type as NodePort.
-func (c *client) serviceClassifierFlows(nodePortAddresses []net.IP, ipProtocol binding.Protocol) []binding.Flow {
-	virtualServiceIP := config.VirtualServiceIPv4
-	if ipProtocol == binding.ProtocolIPv6 {
-		virtualServiceIP = config.VirtualServiceIPv6
-	}
-	// Generate flows for every NodePort IP address. The flows are used to match the first packet of Service NodePort from
-	// Pod.
+// nodePortMarkFlows generates the flows to mark the first packet of Service NodePort connection with ToNodePortAddressRegMark,
+// which indicates the Service type is NodePort.
+func (f *featureService) nodePortMarkFlows(nodePortAddresses []net.IP, ipProtocol binding.Protocol) []binding.Flow {
+	cookieID := f.cookieAllocator.Request(f.category).Raw()
 	var flows []binding.Flow
+	// This generates a flow for every NodePort IP. The flows are used to mark the first packet of NodePort connections
+	// from a local Pod.
 	for i := range nodePortAddresses {
 		flows = append(flows,
-			ServiceClassifierTable.BuildFlow(priorityNormal).
-				Cookie(c.cookieAllocator.Request(cookie.Service).Raw()).
+			NodePortMarkTable.ofTable.BuildFlow(priorityNormal).
+				Cookie(cookieID).
 				MatchProtocol(ipProtocol).
 				MatchDstIP(nodePortAddresses[i]).
 				Action().LoadRegMark(ToNodePortAddressRegMark).
 				Done())
 	}
-	// Generate flow for the virtual IP. The flow is used to match the first packet of Service NodePort from Antrea gateway,
-	// because the destination IP of the packet has already performed DNAT with the virtual IP on host.
+	// This generates the flow for the virtual IP. The flow is used to mark the first packet of NodePort connection from
+	// the Antrea gateway (the connection is performed DNAT with the virtual IP in host netns).
 	flows = append(flows,
-		ServiceClassifierTable.BuildFlow(priorityNormal).
-			Cookie(c.cookieAllocator.Request(cookie.Service).Raw()).
+		NodePortMarkTable.ofTable.BuildFlow(priorityNormal).
+			Cookie(cookieID).
 			MatchProtocol(ipProtocol).
-			MatchDstIP(virtualServiceIP).
+			MatchDstIP(f.virtualIPs[ipProtocol]).
 			Action().LoadRegMark(ToNodePortAddressRegMark).
 			Done())
 
 	return flows
 }
 
-// serviceLearnFlow generates the flow with learn action which adds new flows in
-// sessionAffinityTable according to the Endpoint selection decision.
-func (c *client) serviceLearnFlow(groupID binding.GroupIDType, svcIP net.IP, svcPort uint16, protocol binding.Protocol, affinityTimeout uint16, nodeLocalExternal bool, svcType v1.ServiceType) binding.Flow {
+// serviceLearnFlow generates the flow with learn action which adds new flows in SessionAffinityTable according to the
+// Endpoint selection decision.
+func (f *featureService) serviceLearnFlow(groupID binding.GroupIDType,
+	svcIP net.IP,
+	svcPort uint16,
+	protocol binding.Protocol,
+	affinityTimeout uint16,
+	nodeLocalExternal bool,
+	serviceType v1.ServiceType) binding.Flow {
```

Comment:
nit: we should avoid changes such as changing svcType to serviceType in this PR to avoid distraction by unrelated change like L2249. And there are also svcIP, svcPort, it becomes less unified.

Reply:
Thanks, I'll change it back.